community/cypher/cypher/src/main/scala/org/neo4j/cypher/ExecutionEngine.scala

```diff
@@ -252,6 +253,12 @@ class ExecutionEngine(graph: GraphDatabaseService, logProvider: LogProvider = Nu
       .andThen(_.getOrElse(defaultValue))
       .applyOrElse(graph, (_: GraphDatabaseService) => defaultValue)
   }
+
+  private def makeValueJavaCompatible(value: Any): AnyRef = value match {
```

Comment:
This exact function also exists in `PipeExecutionResult`. Is it possible to reuse that code somehow?